### PR TITLE
scoll/ucc: add onesided ucc support

### PR DESCRIPTION
**Pull request by wfaderhold21**
This PR enables the one-sided collective support recently added in UCC. In addition, the PR delays UCC team creation until the first collective issued by an OSHMEM group in order to delay UCC team creation overhead.

Signed-off-by: ferrol aderholdt <faderholdt@nvidia.com>

### CODE REVIEW

**Comment by jsquyres** [comment]
Does this PR change the minimum required version of UCX?  If so, you need to update the UCX PML configury to require that.

**Comment by jsquyres** [comment]
ok to test

**Comment by jsquyres** [comment]
FYI @open-mpi/ucx

**Comment by wfaderhold21** [comment]
> Does this PR change the minimum required version of UCX? If so, you need to update the UCX PML configury to require that.

This PR does not need to update the minimum version of UCX.

**Comment by wfaderhold21** [comment]
@janjust  Please review.

File: oshmem/mca/scoll/ucc/scoll_ucc_module.c [hunk 1]
```diff
@@ -373,12 +444,12 @@ mca_scoll_ucc_comm_query(oshmem_group_t *osh_group, int *priority)
...
 
     if (!cm->libucc_initialized) {
-        if (0 < cm->nr_modules) {
+        if (memheap_map && memheap_map->n_segments > 0) {
             if (OSHMEM_SUCCESS != mca_scoll_ucc_init_ctx(osh_group)) {
                 cm->ucc_enable = 0;
                 return NULL;
             }
-        }
+        }
```
Comment (janjust) [inline oshmem/mca/scoll/ucc/scoll_ucc_module.c:452]: minor - extra space after }

**Review by janjust — APPROVED** [review]

File: oshmem/mca/scoll/ucc/scoll_ucc_module.c (hunk 1 above)
Comment (wfaderhold21) [reply oshmem/mca/scoll/ucc/scoll_ucc_module.c]: fixed. also added a memset on line 334 of scoll_ucc_module.c as UCC API changed to require this.

**Comment by janjust** [comment]
@wfaderhold21 will you be opening a 5.0 PR?

**Comment by wfaderhold21** [comment]
@janjust done! See #9967

**Pull request merged**